File: php/features.php
<?php
declare(strict_types=1);

// Ctp SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CtpFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CtpBaseFeature();
            case "test":
                return new CtpTestFeature();
            default:
                return new CtpBaseFeature();
        }
    }
}

File: py/features.py
# Ctp SDK feature factory

from feature.base_feature import CtpBaseFeature
from feature.test_feature import CtpTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CtpBaseFeature(),
        "test": lambda: CtpTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
